torch/_dynamo/codegen.py

```diff
@@ -613,6 +613,18 @@ def collect_temp_source(source):
             if arg.source is not None:
                 collect_temp_source(arg.source)
 
+        cm_var = None
+        if config.record_pre_graph_bytecode_in_traces:
+            # Record the pregraph bytecode start
+            self.add_push_null(
+                lambda: self.load_import_from(
+                    utils.__name__, "record_pregraph_bytecode_enter"
+                )
+            )
+            self.extend_output(create_call_function(0, False))
+            cm_var = self.new_var()
+            self.store(cm_var)
```

Comment:
To check my understanding: if the profiler is off, we still call the record_pregraph_bytecode_enter and record_pregraph_bytecode_exit functions but those set up a contextlib.noop, right?